src/main/java/com/puppycrawl/tools/checkstyle/checks/design/DesignForExtensionCheck.java

```diff
@@ -322,10 +343,31 @@ private static boolean branchContainsJavadocComment(DetailAST token) {
             }
             curNode = toVisit;
         }
-
         return result;
     }
 
+    /**
+     * Checks whether a javadoc follows the specified comment pattern.
+     *
+     * @param token tree token
+     * @return true if the javadoc comment of the specified token follows the given pattern
+     */
+    private boolean isValidJavadocComment(DetailAST token) {
+        final Pattern javadocStars = Pattern.compile("\\*\\s*");
+        final Pattern javadocWhiteSpaces = Pattern.compile("\\s+");
+        String javadoc = javadocWhiteSpaces
+                .matcher(javadocStars.matcher(JavadocUtil.getBlockCommentContent(token))
+                .replaceAll(""))
+                .replaceAll(" ");
+
+        final int firstTagIndex = javadoc.indexOf('@');
+        if (firstTagIndex != -1) {
+            javadoc = javadoc.substring(0, firstTagIndex);
```

Comment:
I think we can simply match whole javadoc. 
If javadoc is so sophisticated .... it is already kind of ready to extension, so any extra matching is ok.

in addition it is not very robust way to get description part.